hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/IncrementalContainerReportHandler.java

```diff
@@ -108,9 +108,14 @@ public void onMessage(final IncrementalContainerReportFromDatanode report,
         } catch (ContainerReplicaNotFoundException e) {
           LOG.warn("Container {} replica not found!",
               replicaProto.getContainerID());
-        } catch (NotLeaderException nle) {
-          LOG.warn("Failed to process " + replicaProto.getState()
-              + " Container " + id + " due to " + nle);
+        } catch (SCMException ex) {
+          if (ex.getResult() == SCMException.ResultCodes.SCM_NOT_LEADER) {
+            LOG.warn("Failed to process {} container {}",
+                replicaProto.getState(), id, ex);
```

Comment:
@ashishkumar50 , thanks for working on this!

Please print only the exception message but not the stack trace. #4302 was to avoid printing the stack trace.

Reply:
Updated thanks.